src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -5492,13 +5500,17 @@ finalize_dfs(void)
 			continue;
 		}
 		rc = daos_cont_close(dfs_list[i].coh, NULL);
+		if (rc == -DER_NOMEM)
+			rc = daos_cont_close(dfs_list[i].coh, NULL);
```

Comment:
This isn't desirable.  Since initially writing this test I've added logic to cover this case so that if DER_BUSY is logged then leak checking is disabled.  If we enable the new flag for this test then what would happen here is that dfs_umount() would fail with -DER_NOMEM, then daos_cont_close() and daos_pool_disconnect() would fail with -DER_BUSY and NLT would not run the leak checker in this case so the test would pass without this change.

Reply:
Removed as suggested. Thank you!

Reply:
@ashleypittman @mchaarawi 
		rc = daos_cont_close(dfs_list[i].coh, NULL);
		if (rc != -DER_SUCCESS && rc != -DER_NONEXIST)
			rc = daos_cont_close(dfs_list[i].coh, NULL);

		rc = daos_pool_disconnect(dfs_list[i].poh, NULL);
		if (rc != -DER_SUCCESS && rc != -DER_NONEXIST)
			rc = daos_pool_disconnect(dfs_list[i].poh, NULL);

Shall we retry daos_cont_close() and daos_pool_disconnect when they fail? Is there resource leak on server side without retry on client side?

The fault injections causes client-side memory leak are in tse_task_create(), 
https://github.com/daos-stack/daos/blob/master/src/common/tse.c#L961

Calling stacks,
daos_cont_close() failed
DBG> 0, /scratch/leihuan1/tickets/13548/daos/install/lib64/libgurt.so.4(print_trace+0x67) [0x7f84b1ea0886]
DBG> 1, /scratch/leihuan1/tickets/13548/daos/install/lib64/libgurt.so.4(d_should_fail+0x574) [0x7f84b1ea0ee7]
DBG> 2, /scratch/leihuan1/tickets/13548/daos/install/lib64/libdaos_common.so(tse_task_create+0x5a) [0x7f84b3cb389e]
DBG> 3, /scratch/leihuan1/tickets/13548/daos/install/lib64/libdaos.so.2(dc_task_create+0x80) [0x7f84b3f4e4cd]
DBG> 4, /scratch/leihuan1/tickets/13548/daos/install/lib64/libdaos.so.2(daos_cont_close+0x1dd) [0x7f84b3f2f9df]
DBG> 5, /scratch/leihuan1/tickets/13548/daos/install/lib64/libpil4dfs.so(+0x250ba) [0x7f84b4d5f0ba]
DBG> 6, /scratch/leihuan1/tickets/13548/daos/install/lib64/libpil4dfs.so(+0x23992) [0x7f84b4d5d992]
DBG> 7, /lib64/ld-linux-x86-64.so.2(+0x151e6) [0x7f84b4f8e1e6]
DBG> 8, /lib64/libc.so.6(+0x5123c) [0x7f84b459423c]
DBG> 9, /lib64/libc.so.6(on_exit+0) [0x7f84b4594370]
DBG> 10, /scratch/leihuan1/tickets/13548/daos/install/lib64/libpil4dfs.so(+0x22015) [0x7f84b4d5c015]
DBG> 11, /lib64/libc.so.6(__libc_start_main+0xfa) [0x7f84b457dcfa]
DBG> 12, ls(+0x5e7e) [0x561f33b92e7e]

daos_pool_disconnect() failed
DBG> 0, /scratch/leihuan1/tickets/13548/daos/install/lib64/libgurt.so.4(print_trace+0x67) [0x7f315e085886]
DBG> 1, /scratch/leihuan1/tickets/13548/daos/install/lib64/libgurt.so.4(d_should_fail+0x574) [0x7f315e085ee7]
DBG> 2, /scratch/leihuan1/tickets/13548/daos/install/lib64/libdaos_common.so(tse_task_create+0x5a) [0x7f315fe9889e]
DBG> 3, /scratch/leihuan1/tickets/13548/daos/install/lib64/libdaos.so.2(dc_task_create+0x80) [0x7f31601334cd]
DBG> 4, /scratch/leihuan1/tickets/13548/daos/install/lib64/libdaos.so.2(daos_pool_disconnect+0x1dd) [0x7f316012f3ae]
DBG> 5, /scratch/leihuan1/tickets/13548/daos/install/lib64/libpil4dfs.so(+0x25269) [0x7f3160f44269]
DBG> 6, /scratch/leihuan1/tickets/13548/daos/install/lib64/libpil4dfs.so(+0x23992) [0x7f3160f42992]
DBG> 7, /lib64/ld-linux-x86-64.so.2(+0x151e6) [0x7f31611731e6]
DBG> 8, /lib64/libc.so.6(+0x5123c) [0x7f316077923c]
DBG> 9, /lib64/libc.so.6(on_exit+0) [0x7f3160779370]
DBG> 10, /scratch/leihuan1/tickets/13548/daos/install/lib64/libpil4dfs.so(+0x22015) [0x7f3160f41015]
DBG> 11, /lib64/libc.so.6(__libc_start_main+0xfa) [0x7f3160762cfa]
DBG> 12, ls(+0x5e7e) [0x55b1934ace7e]

Reply:
@ashleypittman @mchaarawi RPC is needed in daos_cont_close() and daos_pool_disconnect(), right? How we should proceed here? It looks retries exist in several other places. Shall I also adopt retry? Thank you!

Reply:
i cannot do a similar "fix" that i did for object close since as you said those are not local operations and require a tse task to be attached to the operation / RPC. 
I also do not think we should add more retries / hacks like this and should work on removing them. I'm not sure what can be done here other than not do fault injection on this path.

Reply:
@mchaarawi I see. Thank you very much for your comments!